snuba/migrations/group_loader.py

```diff
@@ -30,14 +32,46 @@ class DirectoryLoader(GroupLoader, ABC):
     """
     Loads migrations that are defined as files of a directory. The file name
     represents the migration ID.
+
+    Migrations must be named: xxxx_migration_name.py where xxxx is 4 digit,
+    0 padded migration number. As regex: [0-9][0-9][0-9][0-9]_.*\.py
+    Within a dir, migration number are strictly increasing by 1 beginning at
+    0001
     """
 
     def __init__(self, module_path: str) -> None:
-        self.__module = module_path
+        self.__module = module_path  # the one with dots not slashes
 
-    @abstractmethod
     def get_migrations(self) -> Sequence[str]:
-        raise NotImplementedError
+        """
+        Migrations must be in the folder specified by module_path.
+        see class comment for migration naming scheme.
+        """
+        # the folder that the migrations should be in
+        migration_folder = self.__module.replace(".", "/")
+        if not os.path.exists(migration_folder):
+            return []
+        # grab the migrations, ignore all other files
+        files = sorted(
+            filter(
+                lambda f: os.path.isfile(os.path.join(migration_folder, f)),
+                os.listdir(migration_folder),
+            )
+        )
+        migration_filenames = []
+        pattern = re.compile(r"[0-9][0-9][0-9][0-9]_.*\.py")
+        for f in files:
+            if re.fullmatch(pattern, f):
+                migration_filenames.append(f[:-3])
+
+        # validate the migration numbers are strictly increasing starting at 0001
+        for i, fname in enumerate(migration_filenames):
+            if not (int(fname[:4]) == i + 1):
+                raise ValueError(
+                    f"Migrations in folder {migration_folder} were not formatted as expected. Expected migration number {str(i+1).zfill(4)} for migration {fname} but got {fname[:4]}"
+                )
+
+        return migration_filenames
```

Comment:
Should we be caching the results of the migrations ? Or is this called infrequently enough that it wouldn't provide much benefit?

Reply:
I'm not sure what calls this function, but my guess (based on what the function does) would be that it is not called very frequently or under time sensitive circumstances.

Additionally since caching adds complexity, maybe it would be better to leave out unless we find it necessary in the future? Unless you have reason to believe that it may be necessary.